src/mmap.rs

```diff
@@ -444,6 +444,62 @@ impl GuestMemoryMmap {
 
         Ok(Self { regions })
     }
+
+    /// Insert a region into the `GuestMemoryMmap` object and return a new `GuestMemoryMmap`.
+    ///
+    /// # Arguments
+    /// * `region`: the memory region to insert into the guest memory object.
+    pub fn insert_region(
+        &self,
+        region: Arc<GuestRegionMmap>,
+    ) -> result::Result<GuestMemoryMmap, Error> {
+        // It shouldn't be a huge Vec, so prefer simple implementation.
+        for reg in self.regions.iter() {
+            if region.start_addr() >= reg.start_addr() && region.start_addr() <= reg.last_addr() {
+                return Err(Error::MemoryRegionOverlap);
+            } else if reg.start_addr() >= region.start_addr()
+                && reg.start_addr() <= region.last_addr()
+            {
+                return Err(Error::MemoryRegionOverlap);
+            }
+        }
```

Comment:
Since some similar logic is also used in `GuestMemoryMmap::from_arc_regions()` I think it would be worth defining a method that checks if 2 regions overlap inside the `GuestMemoryRegion` trait.

Reply:
Good idea.